pkg/rpc/contractRPC.go

```diff
@@ -135,10 +136,15 @@ func SubmitContractTxn(
 			auth.GasPrice = maxGasPrice
 		}
 
-		level.Info(logger).Log("msg", "gas price", "value", gasPrice)
-
 		wrapper := contractWrapper{auth, account.Address, tellor}
 		tx, err := callback(ctx, wrapper)
+		level.Info(logger).Log("msg", "sent txn on-chain",
```

Comment:
This will panic if tx is nill.
Now looking the logs we can remove it complitely and just add all this data in the log  in handleSubmit
```
level.Info(s.logger).Log("msg", "submited a solution",
 "txHash", tx.Hash().String(),
 "nonce", tx.Nonce(),
 "gasPrice", tx.GasPrice(),
 "data", fmt.Sprintf("%x", tx.Data()),
 "value", tx.Value(),
)
```